lib/badge.rb

```diff
@@ -0,0 +1,75 @@
+class Badge
+  include DSL
+
+  attr_reader :title, :description, :goal, :rank, :user
+
+  def self.slug
+    name.underscore.dasherize.sub('-badge', '')
+  end
+
+  def slug
+    self.class.slug
+  end
+
+  def initialize(user)
+    @user = user
+    describe_context
+  end
+
+  def progress
+    if self.class.progress.nil?
+      nil
+    else
+      @progress ||= instance_eval(&self.class.progress)
+    end
+  end
+
+  def earned?
+    return false unless @goal
+    if progress.nil?
+      instance_eval(&@goal)
+    else
+      progress > @goal
+    end
+  end
```

Comment:
`earned?` return boolean, but `#get_goal_result` not. I think they different. [flog gem](http://ruby.sadi.st/Flay.html) don't find similarities.